examples/tensorflow-akka/src/main/scala/modelserving/TensorFlowModel.scala

```diff
@@ -0,0 +1,167 @@
+package modelserving
+
+import java.io.File
+import java.nio.file._
+
+import scala.collection.JavaConverters._
+import scala.collection.mutable.{ Map ⇒ MMap }
+import scala.util.control.NonFatal
+import scala.util.Try
+
+import com.google.protobuf.Descriptors
+import org.tensorflow.framework._
+import org.tensorflow.{ Graph, SavedModelBundle, Session, Tensor }
+import modelserving.wine.avro._
+
+import TensorFlowModelBundle._
+
+/**
+ * Encapsulates TensorFlow scoring using a [[LoadedModel]] (from a SavedModelBundle).
+ */
+trait TensorFlowModel[Record, ServingResult] {
+  def loadedModel: LoadedModel
+
+  def emptyServingResult: ServingResult
+  def toTensor(record: Record): Tensor[_]
+  def invokeModel(record: Record): Either[String, ServingResult]
+
+  val startTime = System.currentTimeMillis
+
+  /**
+   * Score a record with the model
+   */
+  def score(record: Record): (ServingResult, ModelResultMetadata) = {
+    val start = System.currentTimeMillis()
+    val (errors, modelOutput) = invokeModel(record) match {
+      case Left(errors)  ⇒ (errors, emptyServingResult)
+      case Right(output) ⇒ ("", output)
+    }
+    val duration = (System.currentTimeMillis() - start)
+    val resultMetadata = ModelResultMetadata(errors, loadedModel.modelName, startTime, duration)
```

Comment:
do we need both `start` and `startTime` ?

Reply:
Well... need is a big word. startTime is when we started serving with this model (since start of the streamlet), start is for the duration, how long it took to serve 1..

Reply:
👍